### PR TITLE
Adding in a healthcheck endpoint for the call to the tone analyzer tone_chat endpoint

### PR DESCRIPTION
**Pull request by aprilwebster**
Need to test whether the demo can make a successful request to the Tone Analyzer tone_chat endpoint.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -49,6 +49,20 @@ app.post('/api/tone_chat', (req, res, next) => {
   });
 });
 
+// Endpoint test for call to tone-analyzer
+// if an error is returned from a request to the tone-analyzer tone_chat endpoint,
+// return a 500, otherwise return a 200.
+app.get('/healthchecks/tone_analyzer_request_test', (req, res) => {
```
Comment (germanattanasio) [inline app.js]: just call it `/healthcheck`

Comment (germanattanasio) [inline app.js]: Can you also log the response time?

**Comment by jsstylos** [comment]
Probably want ```    headers: { "X-Watson-Learning-Opt-Out": true
  }``` to be able to filter these out from the service logs.

**Comment by jsstylos** [comment]
We might want something like the upcoming Conversation /logs endpoint API, where we document the request payload, the response payload, request timestamp, response timestamp, for each call.

Something like:

`/health_check`: `{ "logs": [ {"request": {}, "request_timestamp": "...", "response": {}, "response_timestamp": "..."} ]}`

File: app.js (hunk 1 above)
Comment (aprilwebster) [inline app.js]: Sure, can do.

Comment (aprilwebster) [inline app.js]: Done and pushed.

File: config/express.js [hunk 2]
```diff
@@ -28,6 +28,61 @@ module.exports = function (app) {
...
+  // return a 502, otherwise return a 200.
+  app.get('/healthcheck', (req, res) => {
+    const ToneAnalyzerV3 = require('watson-developer-cloud/tone-analyzer/v3');
+    const toneAnalyzer = new ToneAnalyzerV3({
+      // If unspecified here, the TONE_ANALYZER_USERNAME and
+      // TONE_ANALYZER_PASSWORD env properties will be checked
+      // After that, the SDK will fall back to the bluemix-provided VCAP_SERVICES environment property
+      // username: '<username>',
+      // password: '<password>',
+      url: 'https://gateway.watsonplatform.net/tone-analyzer/api',
```
Comment (germanattanasio) [inline config/express.js]: Comment the `url` parameter

File: config/express.js [hunk 3]
```diff
@@ -28,6 +28,61 @@ module.exports = function (app) {
...
+
+      if (err) {
+        return res
+        .status(502)
+        .json({
+          logs: [
+            {
+              request: requestPayload,
+              request_timestamp: requestTimestamp,
+              response: err.toString(),
```
Comment (germanattanasio) [inline config/express.js]: `JSON.stringify(err)`

File: config/express.js (hunk 2 above)
Comment (aprilwebster) [inline config/express.js]: Done.

File: config/express.js (hunk 3 above)
Comment (aprilwebster) [inline config/express.js]: And, done.

**Pull request merged**